relay-metrics/src/aggregation.rs

```diff
@@ -1578,19 +1579,17 @@ impl AggregatorService {
     fn normalize_metric_name(key: &mut BucketKey) -> Result<(), AggregateMetricsError> {
         key.metric_name = match MetricResourceIdentifier::parse(&key.metric_name) {
             Ok(mri) => {
-                if matches!(mri.namespace, MetricNamespace::Unsupported) {
-                    relay_log::debug!("invalid metric namespace {:?}", key.metric_name);
-                    return Err(AggregateMetricsErrorKind::UnsupportedNamespace.into());
-                }
-
                 let mut metric_name = mri.to_string();
                 // do this so cost tracking still works accurately.
                 metric_name.shrink_to_fit();
                 metric_name
             }
             Err(_) => {
-                relay_log::debug!("invalid metric name {:?}", key.metric_name);
-                return Err(AggregateMetricsErrorKind::InvalidCharacters.into());
+                relay_log::debug!("invalid metric namespace {:?}", key.metric_name);
+                return Err(AggregateMetricsErrorKind::UnsupportedNamespace.into());
```

Comment:
Let's unite all parse errors into a new `InvalidIdentifier` kind.

```suggestion
                relay_log::debug!("invalid metric identifier {:?}", key.metric_name);
                return Err(AggregateMetricsErrorKind::InvalidIdentifier.into());
```